### ISSUE TITLE
Type-Checking on Python `@property`

### ISSUE THREAD

**Issue by AK2001** [astral-sh/ty#3816]
### Question

Should type checkers apply on custom Python `@property` "attributes"? I've stumbled upon the following case:

```python
class MyClass:
    def __init__(self, value: int | None = None):
        self.value = value

    @property
    def flag(self) -> bool:
        return value is not None

    def fun(self) -> int:
        if self.flag:
            return value + 10 # Unsupported `+` operationty[unsupported-operator](https://ty.dev/rules#unsupported-operator)
        return 0
``` 

My use case (as seen in the example) revolves around establishing a single boolean value from multiple instance attributes that can be used many times as a short form of writing the entire condition.

The same is observed when `flag` is defined in the class `__init__()` of course, so I guess my question becomes a little bit more complicated. I guess the "solution" would be for the type-checker to look into the operations behind a boolean condition (I'm not sure if this makes sense).

### Version

ty 0.0.51 (b5ef61c13 2026-06-18)

**Comment by AlexWaygood** [comment]
Thanks! Please see https://github.com/astral-sh/ty/issues/3212
